activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -509,6 +509,10 @@ def combine_bind_parameters(
         result
       end
 
+      def default_index_type?(index) # :nodoc:
+        index.using.nil? || index.using == :btree
```

Comment:
sqlite3, oracle-enhanced and sqlserver adapters are always `index.using = nil`.
https://github.com/rsim/oracle-enhanced/blob/7d16086ef0f91fa32a19907c35a97c36bcda4c80/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb#L680
https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/blob/3cee83caa631870f5dcc8dbd89f08da8c30013bc/lib/active_record/connection_adapters/sqlserver/schema_statements.rb#L68
https://github.com/rails/rails/blob/1c968b484bffb2a7b206b47434f7cf61d10e85a2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L359

mysql2 and postgresql adapters are `index.using = :btree` as default.